src/main/kotlin/org/neo4j/graphql/Translator.kt

```diff
@@ -86,12 +86,74 @@ class Translator(val schema: GraphQLSchema) {
                            "WITH $variable as toDelete, ${mapProjection.query} AS $variable $ordering$skipLimit DETACH DELETE toDelete RETURN $variable",
                                 (mapProjection.params + mapOf(paramName to properties.params[paramName])))
                     }
-                    else -> throw IllegalArgumentException("Unknown Mutation "+name)
+                    else -> checkRelationships(fieldDefinition, field, ordering, skipLimit, ctx)
                 }
             }
         }
     }
 
+    private fun checkRelationships(graphQLFieldDefinition: GraphQLFieldDefinition, field: Field, ordering: String, skipLimit: String, ctx: Context): Cypher {
```

Comment:
this is a gigantic method we definitely need to break it up

Reply:
ok